src/openrct2/paint/Paint.cpp

```diff
@@ -18,14 +18,17 @@
 #include "tile_element/Paint.TileElement.h"
 
 #include <algorithm>
+#include <array>
+#include <atomic>
 
 // Globals for paint clipping
 uint8_t gClipHeight = 128; // Default to middle value
 LocationXY8 gClipSelectionA = { 0, 0 };
 LocationXY8 gClipSelectionB = { MAXIMUM_MAP_SIZE_TECHNICAL - 1, MAXIMUM_MAP_SIZE_TECHNICAL - 1 };
 
-paint_session gPaintSession;
-static bool _paintSessionInUse;
+static std::array<paint_session, 512> _paintSessions;
```

Comment:
Do we really want this? Do you realise how big `paint_session` is? 276624 bytes, without taking into account your changes (which grow it by another 16), times 512 it's 141'631'488 bytes or just shy over 135MiB.

Reply:
You would want a paint session per thread and allocating them each time is quite expensive, there might be a better way, with that I agree.